docs/developers/gas-and-fees/user-defined-smart-contracts.md

```diff
@@ -0,0 +1,251 @@
+
+With respect to the [VM Gas Schedule](https://github.com/ElrondNetwork/elrond-config-mainnet/tree/master/gasSchedules), the aforementioned **_on-the-spot_ remaining gas** has to satisfy the following (in order for the **temporary gas lock reservation** to succeed):
+
+```
+onTheSpotRemainingGas > gasToLockForCallback
+
+gasToLockForCallback = 
+    costOf(AsyncCallStep) + 
+    costOf(AsyncCallbackGasLock) + 
+    codeSizeOf(callingContract) * costOf(AoTPreparePerByte)
+```
+
+:::note
+Multiple asynchronous calls employ multiple temporary gas locks, one for each callback function.
+:::
+
+For our example, where `A` has 453 bytes, the `gasToLockForCallback` would be (as of February 2022):
+
+```
+gasToLockForCallback = 100000 + 4000000 + 100 * 453 = 4145300
+```
+
+It follows that the value of `gasLimit` should be: 
+
+```
+simulatedCost < gasLimit < simulatedCost + gasToLockForCallback
+```
+
+For our example, that would be:
+
+```
+3473900 < gasLimit < 7619200
+```
+
+:::important
+As of February 2022, for contracts that call other contracts, the lowest `gasLimit` required by a successful execution isn't easy to determine using **erdpy**. While this value can be determined by a careful inspection of the local testnet logs, for the moment, the recommended approach is to **start with the right-hand side of the inequality above (`simulatedCost + gasToLockForCallback`) and gradually decrease the value** while simulating the call and looking for a successful output.
+:::
+
+For our example, let's simulate using the following values for `gasLimit`: `7619200, 7000000, 6000000`:
+
+```
+$ erdpy --verbose contract call erd1qqqqqqqqqqqqqpgqfzydqmdw7m2vazsp6u5p95yxz76t2p9rd8ss0zp9ts \
+ --pem=./testnet/wallets/users/alice.pem \
+ --function=foo\
+ --recall-nonce --gas-limit=7619200\
+ --arguments ${hexAddressOfB}\
+ --simulate
+
+... inspect output (possibly testnet logs); execution is successful
+
+erdpy --verbose contract call erd1qqqqqqqqqqqqqpgqfzydqmdw7m2vazsp6u5p95yxz76t2p9rd8ss0zp9ts \
+ --pem=./testnet/wallets/users/alice.pem \
+ --function=foo\
+ --recall-nonce --gas-limit=7000000\
+ --arguments ${hexAddressOfB}\
+ --simulate
+
+... inspect output (possibly testnet logs); execution is successful
+
+erdpy --verbose contract call erd1qqqqqqqqqqqqqpgqfzydqmdw7m2vazsp6u5p95yxz76t2p9rd8ss0zp9ts \
+ --pem=./testnet/wallets/users/alice.pem \
+ --function=foo\
+ --recall-nonce --gas-limit=6000000\
+ --arguments ${hexAddressOfB}\
+ --simulate
+
+... inspect output (possibly testnet logs); ERROR: out of gas when executing B::bar()
+
+```
+
+Therefore, in our case, a reasonable value for `gasLimit` would be 7000000.
+
+:::important
+In case of a highly gas-demanding callback (not recommended) which would consume more than `gasToLockForCallback`, one should appropriately increase the right-hand side of the `gasLimit` inquality depicted above, by inspecting the contract call output and the testnet logs.
```

Comment:
inequality*

Reply:
Inequation, because it can change with the value of its variables. An inequality would remain true regardless of the values of the variables.

But nobody ever cares about this distinction anyway... At least not in English, anyway. Not even Wikipedia.

Reply:
Fixed.